apps/03TicTacToe.js

```diff
@@ -1,88 +1,122 @@
-    console.log('0 ' + board[0].join(' | '));
-    console.log('  ---------');
-    console.log('1 ' + board[1].join(' | '));
-    console.log('  ---------');
-    console.log('2 ' + board[2].join(' | '));
-}
-
-function horizontalWin() {
-    // Your code here
-}
-
-function verticalWin() {
-    // Your code here
-}
-
-function diagonalWin() {
-    // Your code here
-}
-
-function checkForWin() {
-    // Your code here
-}
-
-function ticTacToe(row, column) {
-    // Your code here
-}
-
-function getPrompt() {
-    printBoard();
-    console.log("It's Player " + playerTurn + "'s turn.");
-    prompt.get(['row', 'column'], function (error, result) {
-        ticTacToe(result['row'], result['column']);
-        getPrompt();
-    });
-}
-
-
-
-// Tests
-
-if (typeof describe !== 'undefined') {
-
-    describe('#ticTacToe()', function () {
-        it('should place mark on the board', function () {
-            ticTacToe(1, 1);
-            assert.deepEqual(board, [ [' ', ' ', ' '], [' ', 'X', ' '], [' ', ' ', ' '] ]);
-        });
-        it('should alternate between players', function () {
-            ticTacToe(0, 0);
-            assert.deepEqual(board, [ ['O', ' ', ' '], [' ', 'X', ' '], [' ', ' ', ' '] ]);
-        });
-        it('should check for vertical wins', function () {
-            board = [ [' ', 'X', ' '], [' ', 'X', ' '], [' ', 'X', ' '] ];
-            assert.equal(verticalWin(), true);
-        });
-        it('should check for horizontal wins', function () {
-            board = [ ['X', 'X', 'X'], [' ', ' ', ' '], [' ', ' ', ' '] ];
-            assert.equal(horizontalWin(), true);
-        });
-        it('should check for diagonal wins', function () {
-            board = [ ['X', ' ', ' '], [' ', 'X', ' '], [' ', ' ', 'X'] ];
-            assert.equal(diagonalWin(), true);
-        });
-        it('should detect a win', function () {
-            assert.equal(checkForWin(), true);
-        });
-    });
-} else {
-
-    getPrompt();
-
-}
+    'use strict';
+
+    var assert = require('assert');
+    var prompt = require('prompt');
+    prompt.start();
+
+    var board = [
+        [' ', ' ', ' '],
+        [' ', ' ', ' '],
+        [' ', ' ', ' ']
+    ];
+
+    var playerTurn = 'X';
+
+    function printBoard() {
+        console.log('   0  1  2');
+        console.log('0 ' + board[0].join(' | '));
+        console.log('  ---------');
+        console.log('1 ' + board[1].join(' | '));
+        console.log('  ---------');
+        console.log('2 ' + board[2].join(' | '));
+    };
+
+    function horizontalWin() {
+        // top horizontal row
+        return (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn ||
+            // middle horizontal row
+            board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn ||
+            // bottom horizontal row
+            board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)
```

Comment:
This is very interesting! I actually never thought to do it this way. So there is no if, and the return will just return the end  result, true/false? If possible could you show me this again during class? I would like to see it!

Awesome :)!
![leo](http://i.imgur.com/M1nEp7h.gif)